resources/patches/media_avportal-improve-exception-management.patch

```diff
@@ -0,0 +1,35 @@
+diff --git a/includes/media_avportal.formatters.inc b/includes/media_avportal.formatters.inc
+index 33638f8..544f8ba 100644
+--- a/includes/media_avportal.formatters.inc
++++ b/includes/media_avportal.formatters.inc
+@@ -53,9 +53,14 @@ function media_avportal_file_formatter_video_view($file, $display) {
+     // Let's add the language to the options.
+     global $language;
+     $wrapper = file_stream_wrapper_get_instance_by_uri($file->uri);
+-    $languages = $wrapper->getAvailableVideoLanguage();
++    try {
++      $languages = $wrapper->getAvailableVideoLanguage();
++    }
++    catch (Exception  $e) {
++      watchdog_exception('media_avportal', $e);
++    }
+     $video_language = 'en';
+-    if (in_array($language->language, $languages)) {
++    if (is_array($languages) && in_array($language->language, $languages)) {
+       $video_language = $language->language;
+     }
+     $options['lg'] = $video_language;
+@@ -113,7 +118,12 @@ function media_avportal_file_formatter_image_view($file, $display) {
+   $wrapper = file_stream_wrapper_get_instance_by_uri($file->uri);
+   $image_style = $display['settings']['image_style'];
+   $valid_image_styles = image_style_options(FALSE);
+-  $path = $wrapper->getLocalThumbnailPath();
++  try {
++    $path = $wrapper->getLocalThumbnailPath();
++  }
++  catch (Exception  $e) {
++    watchdog_exception('media_avportal', $e);
```

Comment:
That's nice to have an exception, but here you don't have a $path variable
That's why i was suggesting to use the default media thumbnail for this.
It should be located at /sites/default/files/media-icons/default/video-x-generic.png and can be accessed using :
public://media-icons

Otherwise we'd still have a fatal error if ws is not available

Reply:
I've added the default media thumbnail and I've removed the exception. There is already an exception inside getLocalThumbnailPath() in which the default thumbnail would be used in case the thumbnail is not available.

Reply:
Ok good catch, so that's fine ! I'll make a new test now !